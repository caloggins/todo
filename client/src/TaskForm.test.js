import React from 'react'
import Container, { TaskForm } from './TaskForm'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'

describe('TaskForm tests...', () => {
    describe('Component rendering tests...', () => {
        it('renders correctly', () => {
            const tree = shallow(<TaskForm />)
            expect(tree).toMatchSnapshot()
        })
    })

    describe('Container tests...', () => {
        const mockStore = configureStore()
        let tree, store

        beforeEach(() => {
            store = mockStore()
            store.dispatch = jest.fn()
            tree = mount(<Container store={store} />)
        })

        it('it does not allow the default action', () => {
            const preventDefault = jest.fn()

            tree.find('form').simulate('submit', { preventDefault })

            expect(preventDefault).toHaveBeenCalled()
        })

        it('it creates an event', () => {
            const preventDefault = jest.fn()

            tree.find('form').simulate('submit', { preventDefault })

            expect(store.dispatch).toHaveBeenCalled()
        })
    })
})
