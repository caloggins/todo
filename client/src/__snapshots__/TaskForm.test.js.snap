// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm tests... Component rendering tests... renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="taskDescription"
    >
      Description:
    </label>
    <input
      className="form-control"
      id="taskDescription"
      onChange={[Function]}
      placeholder="Enter description."
    />
    <button
      className="btn btn-primary"
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
